### PR TITLE
Add new option to change documentation behaviour

### LINKED ISSUE #98: Is there any way to disable preview on completion?
Title.

### PR DESCRIPTION
**Pull request by sblumentritt**
This resolves #98

### CODE REVIEW

**Comment by elianiva** [comment]
wouldn't it be better as `show_documentation` ? Because nvim-compe only shows the documentation on a floating window anyway.

**Comment by hrsh7th** [comment]
Yes. I think option name should be changed.
I prefer `documentation = 'enable' || 'disable'`.

File: lua/compe/completion.lua [hunk 1]
```diff
@@ -95,18 +95,20 @@ end
 
 --- select
 Completion.select = function(index)
-  if index == -1 then
-    return
-  end
+  if Config.get().show_documentation_as_float then
+    if index == -1 then
```
Comment (hrsh7th) [inline lua/compe/completion.lua:99]: This check should be moved to L107.

We must set `_selected_item` even if documentation disabled.

**Comment by cloggier** [comment]
This. There might be future options added later and then having `documentation` could become a table with more specific options.

**Comment by sblumentritt** [comment]
I would prefer the `show_documentation` with `true || false` as boolean checks are more performant than string checks.

**Comment by hrsh7th** [comment]
We must ignore minor optimization in design decisions, I think.

If the user wants to disable documentation only when `preselect`, the boolean flags can't extend for it.

File: lua/compe/completion.lua [hunk 2]
```diff
@@ -103,10 +103,12 @@ Completion.select = function(index)
   if completed_item then
     Completion._selected_item = completed_item
 
-    for _, source in ipairs(Completion.get_sources()) do
-      if source.id == completed_item.source_id then
-        source:documentation(completed_item)
-        break
+    if Config.get().documentation == 'enable' then
```
Comment (hrsh7th) [inline lua/compe/completion.lua:106]: Thank you! It's correct!

**Comment by cloggier** [comment]
I am with @sblumentritt here. Should be bool to start with. More detailed configuration can be part of table this option could accept.

**Comment by hrsh7th** [comment]
LGTM!
Thank you for your contribution! 😄

**Comment by hrsh7th** [comment]
@cloggier 
Is this what you are mean?

```
let g:compe = {}
let g:compe.documentation = v:true || {
\   'show': 'preselect',
\   'layout': 'docked',
\ }
```

**Comment by hrsh7th** [comment]
```
let g:compe = {}
let g:compe.documentation = v:true
```

```
let g:compe = {}
let g:compe.documentation = {}
let g:compe.documentation.disabled = v:true
```

**Comment by cloggier** [comment]
Yes, something like that. Same as sources can be true or a table for more specific configuration.

**Comment by hrsh7th** [comment]
You are absolutely right. I'm wrong. sorry!!

@sblumentritt Really sorry! I had the wrong opinion. Could you change to the above format?

**Comment by sblumentritt** [comment]
No problem 😄 

Do you want the `disabled` as boolean entry or should we use the `show` entry with `'enable' || 'disable'` as string entry?

**Comment by hrsh7th** [comment]
Currently, it is ok to be just string flag.

File: README.md [hunk 3]
```diff
@@ -123,6 +126,10 @@ require'compe'.setup {
   max_kind_width = 100;
   max_menu_width = 100;
 
+  documentation = {
+    show = 'enable';
+  };
```
Comment (hrsh7th) [inline README.md:131]: Sorry for my poor English.

My intention is we should provide only `let g:compe.documentation = v:true | v:false` for now.

**Comment by hrsh7th** [comment]
LGTM. Sorry for the confusion.
Thank you very much!

**Comment by bluz71** [comment]
Many thanks to those you implemented and reviewed this.

This is a very useful option since some of us are minimalists.

nvim-compe keeps getting better.

**Pull request merged**